src/django/pfb_analysis/filters.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+from rest_framework import filters
+import django_filters
+
+from .models import AnalysisJob, AnalysisJobStatusUpdate, Neighborhood
+
+
+logger = logging.getLogger(__name__)
+
+
+class AnalysisJobFilterSet(filters.FilterSet):
+    """ Filters for AnalysisJob:
+      - status, to handle job status being defined as a property and not as database field
+      - latest, to return only the most recent analysis job for each neighborhood
+    """
+
+    status = django_filters.ChoiceFilter(choices=AnalysisJob.Status.CHOICES,
+                                         method='filter_status')
+    latest = django_filters.BooleanFilter(method='filter_latest')
+
+    def filter_status(self, queryset, name, value):
+        if value:
+            matches = [m.pk for m in queryset.all() if m.status == value]
+            queryset = queryset.filter(pk__in=matches)
+
+        return queryset
+
+    def filter_latest(self, queryset, name, value):
+        """ Filters down to the latest successful analysis for each neighborhood, but falls back
+        to the latest modified if there are no successful analysis jobs for a neighborhood.
+
+        This means that if it's applied on top of a status filter, it follows the fallback
+        path and returns the latest job with the given status.
+
+        Runs tons of queries, but the query expression required to do this would be gnarly.
```

Comment:
Definitely would be up for another run at it at some point, since the same logic would probably enable moving the status filter into the query as well, but yeah, it seems reasonable to settle for this for the moment.